src/modules/Channel/context/hooks/useFileInfoListWithUploaded.ts

```diff
@@ -0,0 +1,20 @@
+import { SendingStatus } from '@sendbird/chat/message';
+import { CoreMessageType } from '../../../../utils';
+import { UploadedFileInfoWithUpload } from '../../../../types';
+
+export const useFileInfoListWithUploaded = (message: CoreMessageType): UploadedFileInfoWithUpload[] => {
+  if (!message || !message.isMultipleFilesMessage || !message.isMultipleFilesMessage()) return [];
+  else if (message.sendingStatus === SendingStatus.SUCCEEDED) {
+    return message.fileInfoList.map((it) => ({
+      ...it,
+      url: it.url,
+      isUploaded: true,
+    }));
+  } else {
+    return message.messageParams.fileInfoList.map((it) => ({
+      ...it,
+      url: it.fileUrl ?? (it.file instanceof Blob ? URL.createObjectURL(it.file) : undefined),
+      isUploaded: !it.file && typeof it.fileUrl === 'string' && it.fileUrl.length > 0,
```

Comment:
with this condition, there are two possible scenarios
1. file was given -> was uploaded
2. fileUrl was given -> was uploaded (not really uploaded)

if UploadedFileInfoWithUpload.isUploaded is meant for both scenarios, then no problem.

Reply:
I have adopted the logic from the existing code and understood it as follows:

When a file is provided, if it is uploaded, I understand that the given file is deleted, and the fileUrl is updated.
When fileUrl is provided, since no upload is performed, We treat it as if it has been uploaded.